### PR TITLE
Progress. Chart for classes by subject.

### PR DESCRIPTION
**Pull request by NatalyaTretyak**

### CODE REVIEW

File: src/app/progress/progress/progress.component.ts [hunk 1]
```diff
@@ -171,7 +178,21 @@ export class ProgressComponent implements OnInit {
     }
   }
 
-  averageStudentMark(servData) {
+  getAverageStudentMark(servData) {
+    let avrStudentMark = 0;
+
+    // find average student mark by subject
+    if (servData.length !== 0) {
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: no need to check length, arrays methods will not be run if length is 0

File: src/app/progress/progress/progress.component.ts [hunk 2]
```diff
@@ -171,7 +178,21 @@ export class ProgressComponent implements OnInit {
...
+      servData.forEach((value: any) => {
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: why dont you use .find?

File: src/app/progress/progress/progress.component.ts [hunk 3]
```diff
@@ -171,7 +178,21 @@ export class ProgressComponent implements OnInit {
...
+        if (value.subjectId === this.selectedSubjects.subjectId) {
+          avrStudentMark = value.avgMark;
+        }
+      });
+    }
+    return avrStudentMark;
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: return servData.find(item => item.subjectId === this.selectedSubjects.subjectId).avgMark || 0;

File: src/app/progress/progress/progress.component.ts (hunk 1 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: corrected it

File: src/app/progress/progress/progress.component.ts (hunk 2 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: changed code

File: src/app/progress/progress/progress.component.ts (hunk 3 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: changed code

**Pull request merged**